indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -0,0 +1,138 @@
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request, jsonify, session
+
+from indico.core.db import db
+from indico.modules.events.registration import logger
+from indico.modules.events.registration.controllers.management.sections import RHManageRegFormSectionBase
+from indico.modules.events.registration.models.items import RegistrationFormText, RegistrationFormItem
+from indico.modules.events.registration.models.registration_form_fields import (RegistrationFormField,
+                                                                                RegistrationFormFieldData)
+from indico.util.string import snakify
+from indico.web.util import jsonify_data
+from uuid import uuid4
+
+NON_VERSIONED_DATA = {'min_value', 'length', 'number_of_columns', 'number_of_rows', 'places_limit', 'date_format',
+                      'with_extra_slots', 'input_type', 'default_item', 'time_format'}
+
+
+def _fill_form_field_with_data(field, field_data):
+    field.title = field_data.pop('caption')
+    field.description = field_data.pop('description', '')
+    field.is_enabled = not field_data.pop('disabled')
+    field.is_required = field_data.pop('mandatory', False)
+    field.input_type = field_data.pop('input')
+    field.data = {key: field_data.pop(key) for key in NON_VERSIONED_DATA if key in field_data}
+
+
+class RHManageRegFormFieldBase(RHManageRegFormSectionBase):
+    """Base class for a specific field within a registration form"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.field
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageRegFormSectionBase._checkParams(self, params)
+        self.field = RegistrationFormItem.get_one(request.view_args['field_id'])
+
+
+class RHRegistrationFormToggleFieldState(RHManageRegFormFieldBase):
+    """Enable/Disable a field"""
+
+    def _process(self):
+        self.field.is_enabled = (request.args.get('enable') == 'true')
+        db.session.flush()
+        logger.info('Field {} modified by {}'.format(self.field, session.user))
+        return jsonify_data(**self.field.view_data)
+
+
+class RHRegistrationFormModifyField(RHManageRegFormFieldBase):
+    """Remove/Modify a field"""
+
+    def _process_DELETE(self):
+        self.field.is_deleted = True
+        db.session.flush()
+        logger.info('Field {} deleted by {}'.format(self.field, session.user))
+        return jsonify_data(success=True)
+
+    def _process_POST(self):
+        field_data = request.json['fieldData']
+        _fill_form_field_with_data(self.field, field_data)
```

Comment:
What's the point of returning the same object again? It's something already in the database, it'll always be the same one